[ai assistant] Update product status for 8.12, add known issues (#3512)
elastic/observability-docs: docs/en/observability/observability-ai-assistant.asciidoc

[IMPORTANT]
====
Users should leverage the Observability AI Assistant sensibly as the reliability of its responses might vary.

> Always cross-verify any returned advice for accurate threat detection and response, insights, and query generation.

Author: Discussed in Slack. I used the text provided there.

Knowledge base articles created before 8.12 use ELSER v1.
In 8.12, knowledge base articles must use ELSER v2.
You can either:

* Clear all old knowledge base articles manually and reindex them.
* Upgrade all knowledge base articles indexed with ELSER v1 to ELSER v2 using a https://github.com/elastic/elasticsearch-labs/blob/main/notebooks/model-upgrades/upgrading-index-to-use-elser.ipynb[Python script].
====

The AI Assistant uses {ml-docs}/ml-nlp-elser.html[ELSER], Elastic's semantic search engine, to recall data from its internal knowledge base index to create retrieval augmented generation (RAG) responses. Adding data such as Runbooks, GitHub issues, internal documentation, and Slack messages to the knowledge base gives the AI Assistant context to provide more specific assistance.

NOTE: Your AI provider may collect telemetry when using the AI Assistant. Contact your AI provider for information on how data is collected.

You can add information to the knowledge base by asking the AI Assistant to remember something while chatting (for example, "remember this for next time"). The assistant will create a summary of the information and add it to the knowledge base.

> You can also add external data to the knowledge base either in {kib} using the Stack Management UI or using the {es} Index API.

Reviewer: Should we include a URL to the specific stack management page?

Also we using Serverless the Stack Management UI is not available, but many of its features including the Ai assistant ones are available in the Project Settings UI. Should we include this here?

Author: >Should we include a URL to the specific stack management page?

I don't think we can add a URL because it's specific to each deployment, right? 

>Also we using Serverless the Stack Management UI is not available, but many of its features including the Ai assistant ones are available in the Project Settings UI. Should we include this here?

I'll open a separate PR to update the serverless docs once this PR is finalized! I'll be sure to update the language that describes to the UI to reflect what exists in serverless.

Reviewer: The base for the URL will be different for every deployment, but the last part will be the same.

Stack management: 
* app/management

AI Assistant Management:
* app/management/kibana/aiAssistantManagementSelection

AI assistant for Observability
* app/management/kibana/aiAssistantManagementObservability

Reviewer 2: >Should we include a URL to the specific stack management page

My two cents: As far as I know, we don't typically document the URL. Instead we document the navigation steps. Because URL changes are subtle, we are less likely to notice them. When the navigation steps change, we'll do global updates to fix the problem.

[discrete]
[[obs-ai-stack-management]]
=== Use the UI

To add external data to the knowledge base in {kib}:

> . Go to *Stack Management*.

Reviewer: Or Project Settings when using Serverless

> . In the _Kibana_ section, click *AI Assistants*.

Reviewer: "Then select the Observability AI Assistant"

Author: ```suggestion
. In the _Kibana_ section, click *AI Assistants*.
. Then select the *Elastic AI Assistant for Observability*.
```

[[obs-ai-add-data]]
== Add data to the AI Assistant knowledge base

[IMPORTANT]
====
*If you started using the AI Assistant in technical preview*,

> any knowledge base articles you created before 8.12 will no longer be supported.

Reviewer: Suggestion: 'will have to be reindexed or upgraded before they can be used'

Author: ```suggestion
any knowledge base articles you created before 8.12 will have to be reindexed or upgraded before they can be used.
```

[[obs-ai-interact]]
== Interact with the AI Assistant

> You can chat with the AI Assistant or interact with contextual prompts located throughout {observability}.

Reviewer: Contextual Insights

Author: ```suggestion
You can chat with the AI Assistant or interact with contextual insights located throughout {observability}.
```

image::images/obs-ai-logs.gif[Observability AI assistant example, 75%]

You can continue a conversation from a contextual prompt by clicking *Start chat* to open the AI Assistant chat.

[discrete]
[[obs-ai-known-issues]]
== Known issues

[discrete]
[[obs-ai-token-limits]]
=== Token limits

> Most LLMs have set number of tokens they can manage in single a conversation.

Reviewer 2: ```suggestion
Most LLMs have a set number of tokens they can manage in single a conversation.
```
